### ISSUE TITLE
Increment problem

### ISSUE THREAD

**Issue by feitianstyle** [arangodb/arangodb#4296]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [ ] 3.2
- [ *] 3.3
- [ ] self-compiled devel branch

Mode:
- [ *] Cluster
- [ ] Single-Server

Storage-Engine:
- [ ] mmfiles
- [* ] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [* ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [* ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

I have two collections,These two collections form a graph,I used the connectedComponent algorithm,I divided the vertices into the community,It's convenient for me to finish the business better.But I have added nodes and edges to add to the graph,Is there a way to put the new node into the corresponding community through the side of the relationship, or the new community.Of course I can't use arangodb's own connectedComponent , because I have too much data.I used the graphx algorithm to divide the previous community,So what should I do to solve this problem?

**Comment by Simran-B** [comment]
I can not answer this directly, but have you looked into AQL's UPSERT statement?

What is allows you to do is to check if a certain document exists (can be multiple attributes that are checked to determine the existence), and if it does, update it with some values, and it not, create a new document with other values.

Or maybe it's as simple as fetching a neighbor vertex and add its community ID to the new vertex? I can't show you how to do that however, without knowing how you intent to create new vertices.

**Comment by feitianstyle** [comment]
I import the vertices and edges with arangoimp,No community ID before the vertex import，I just want to have a way to determine which community to add to the new vertices.If the new node can not find the corresponding community in the graph, a new community ID is allocated.If you can meet my needs by other ways of importing, that's OK.Is there any way to do it？@Simran-B

**Comment by Simran-B** [comment]
Communities can not be calculated as part of the import with arangoimp I'm afraid. You will have to process the data once it is in the system. If all existing data has correct community IDs, it would be fairly simple to copy that ID over to a new vertex if you know to which other vertices it will be linked to.

**Comment by feitianstyle** [comment]
Yes, using arangoimp can not calculate the data,Can only be calculated after the data is imported.The community ID for all existing data is correct，I do not know which community the new node should be in,I can import the data into a temporary collection, through the calculation into the existing collection,
But I do not know how to finish it yet,Can you help me?Can it be calculated from the existing set of edges?

**Comment by Simran-B** [comment]
You could import new vertices into a temporary collection indeed. From there, you could iterate over them and create the required vertices in the target collection along with the edges to link these vertices. That means you need to know which existing vertices you want to connect with the new vertices (or vice versa). Then it's only a matter of looking up the community ID of the existing vertex, e.g. with the DOCUMENT() function, and adding it to the data of the new vertex.

Where does your information about what to link the new vertices with come from?

**Comment by feitianstyle** [comment]
First, it is assumed that the newly added vertices have edges and are in the edge set.And then I'll go through the collection and find the community ID,If there are no new vertex edge set, and I will give it to create a new community ID,If you have a new edge, put it directly into the edge set without having to think about the community ID.So how do I go to the community ID?@Simran-B

**Comment by Simran-B** [comment]
Does that mean you import new vertices as well as new edges that link (some) new vertices to existing vertices?

The _from and _to attributes of those edges store the vertex names, of which one is a new vertex and the other references an existing vertex. Fetch the existing vertex data with something like `LET existingVertex = DOCUMENT(edge._to)` to gain access to the stored community ID, merge it into the new vertex with `MERGE(newVert, {yourCommunityAttribute: existingVertex.yourCommunityAttribute})` and update/replace the new vertex (or if you read them from a temporary collection, INSERT the merge result into the target collection).

**Comment by feitianstyle** [comment]
Thank you. I'm trying to write the script with pyarango now,But I came up with a problem when I was importing data.I use arangoimp to import vertex and edges, respectively.It's normal when I import the side,There was a mistake when I introduced the vertex: WARNING {httpclient} ignoring HTTP response with 'Content-Length' bigger than max packet size (363879063 > 134217728)  ，The side has not been imported yet.Then the loop is introduced to the first vertex file,@Simran-B
